common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt

```diff
@@ -173,7 +173,10 @@ class HTTPClient(
             throw IOException(NetworkStrings.HTTP_RESPONSE_PAYLOAD_NULL)
         }
 
-        val verificationResult = if (shouldSignResponse && nonce != null) {
+        val verificationResult = if (shouldSignResponse &&
+            nonce != null &&
+            RCHTTPStatusCodes.isSuccessful(responseCode)
+        ) {
```

Comment:
Note that this means for 400/500s we will return `NOT_VERIFIED` as verification result. I believe this shouldn't matter since the code that handles errors later on will not let it be handled as a success, but I thought I would mention it.